### DISCUSSION TITLE
How to add legends

### DISCUSSION THREAD

**Discussion by MananJethwani** [xyflow/xyflow#2206] [New Features]
Hey, I have nodes in multiple colors and I wanted to add a legend to show which nodes mean what exactly,
Is there a feature I can use to add such legends?

**Comment by bcakmakoglu** [comment]
Well... it's not a built-in feature and doesn't have to be.
You can just create your own sidebar/tooltips or whatever you want, that can display some sort of "legend" for your nodes.

[Here's an example using a simple sidebar.](https://codesandbox.io/s/sandpack-project-forked-2xf87s?file=/App.js)

**Reply by Krithika3** [reply]
@bcakmakoglu how do you create tooltips with this?
